test/acme_server/nonce_test.exs

```diff
@@ -0,0 +1,20 @@
+defmodule AcmeServer.NonceTest do
+  use ExUnit.Case, async: true
+
+  test "new" do
+    first_nonce = AcmeServer.Nonce.new()
+    nonce = AcmeServer.Nonce.new()
+    assert nonce == first_nonce + 1
+  end
+
+  test "verify ok" do
+    nonce = AcmeServer.Nonce.new()
+    assert :ok == AcmeServer.Nonce.verify!(nonce)
+  end
+
+  test "verify not ok" do
+    assert_raise MatchError, fn ->
+      AcmeServer.Nonce.verify!(-1)
```

Comment:
Maybe use something like `:invalid_nonce` instead of magical `-1` (which under some implementation might be a valid nonce)?